test/quic_txp_test.c

```diff
@@ -1535,18 +1534,80 @@ static int run_script(const struct script_op *script)
 
     testresult = 1;
 err:
+    if (!testresult)
+        TEST_error("script %d failed at op %zu", script_idx + 1, opn + 1);
     if (have_helper)
         helper_cleanup(&h);
     return testresult;
 }
 
 static int test_script(int idx)
 {
-    return run_script(scripts[idx]);
+    if (idx + 1 != 18) return 1;
+    return run_script(idx, scripts[idx]);
+}
+
+/*
+ * Dynamic Script 1.
+ *
+ * This script exists to test the interactions between multiple packets (ELs) in
+ * the same datagram when there is a padding requirement (due to the datagram
+ * containing an Initial packet). There are boundary cases which are difficult
+ * to get right so it is important to test this entire space. Examples of such
+ * edge cases include:
+ *
+ * - If we are planning on generating both an Initial and Handshake packet in a
+ *   datagram ordinarily we would plan on adding the padding frames to meet the
+ *   mandatory minimum size to the last packet in the datagram (i.e., the
+ *   Handshake packet). But if the amount of room remaining in a datagram is
+ *   e.g. only 3 bytes after generating the Initial packet, this is not
+ *   enough room for another packet and we have a problem as having finished
+ *   the Initial packet we have no way to add the necessary padding.
+ *
+ * - If we do have room for another packet but it is not enough room to encode
+ *   any desired frame.
+ *
+ * This test confirms we handle these cases correctly for multi-packet datagrams
+ * by placing two packets in a datagram and varying the size of the first
+ * datagram.
+ */
+static const unsigned char dyn_script_1_crypto_1a[1200];
+static const unsigned char dyn_script_1_crypto_1b[1];
+
+static struct script_op dyn_script_1[] = {
+    OP_PROVIDE_SECRET(QUIC_ENC_LEVEL_INITIAL, QRL_SUITE_AES128GCM, secret_1)
+    OP_PROVIDE_SECRET(QUIC_ENC_LEVEL_HANDSHAKE, QRL_SUITE_AES128GCM, secret_1)
+    OP_TXP_GENERATE_NONE()
+    OP_CRYPTO_SEND(QUIC_PN_SPACE_INITIAL, dyn_script_1_crypto_1a) /* [crypto_idx] */
+    OP_CRYPTO_SEND(QUIC_PN_SPACE_HANDSHAKE, dyn_script_1_crypto_1b)
+    OP_TXP_GENERATE()
+    OP_RX_PKT()
+    OP_EXPECT_DGRAM_LEN(1200, 1200)
```

Comment:
Does this check that the HANDSHAKE packet is generated actually - at least when it fits? If I understand this right if TXP just padded the INITIAL packet instead of bothering generating the handshake packet even if it fits, it would still pass?